<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Basic analyses and heatmaps with dlptools</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Basic analyses and heatmaps with
dlptools</h1>



<div id="dlptools" class="section level1">
<h1>dlptools</h1>
<p>dlptools will help with basic file manipulation of DLP pipeline out
and adding of genomic features to read bins for filtering and
classification of changes. Also included are a few standard plots that
are commonly made.</p>
<div id="installation" class="section level2">
<h2>Installation</h2>
<p>You can install the development version of dlptools from <a href="https://github.com/">GitHub</a> with:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="co"># install.packages(&quot;devtools&quot;)</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;molonc/dlptools&quot;</span>)</span></code></pre></div>
</div>
<div id="common-analysis-tasks" class="section level2">
<h2>Common Analysis Tasks</h2>
<p>These are a few steps that are often taken to start a DLP
analysis.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">library</span>(fs)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="fu">library</span>(dlptools)</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="co"># a file path to where some directories with DLP data lives. Expected structure</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="co"># discussed in README.md</span></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a>dlp_dir <span class="ot">&lt;-</span> <span class="st">&quot;/projects/molonc/scratch/bfurman/dlp_testdata/&quot;</span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a><span class="co"># looks something like this</span></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a>fs<span class="sc">::</span><span class="fu">dir_tree</span>(dlp_dir, <span class="at">recurse =</span> <span class="dv">1</span>)</span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a><span class="co">#&gt; /projects/molonc/scratch/bfurman/dlp_testdata/</span></span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a><span class="co">#&gt; └── SC-8408</span></span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a><span class="co">#&gt;     ├── annotation</span></span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a><span class="co">#&gt;     └── hmmcopy</span></span></code></pre></div>
<p>First step is usually to import the cell specific metrics and mark
which cells are good ones to use for downstream analyses.</p>
<p>This example uses some typical thresholds, but you should think about
these in relation to your project and sequence data at hand.</p>
<p>Importing data:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>metrics <span class="ot">&lt;-</span> dlptools<span class="sc">::</span><span class="fu">import_dlp_files</span>(dlp_dir, <span class="st">&quot;metrics&quot;</span>)</span></code></pre></div>
<p>or really, however you like to load a directory of tsvs, this is just
a convenience function as DLP output tends not to change run to run.</p>
<p>Here we can set some thresholds to define “good cells”. These here
are just a suggestion with some commonly used thresholds. These could be
project specific.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>metrics <span class="ot">&lt;-</span> metrics <span class="sc">|&gt;</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>    <span class="at">good_cell =</span> dplyr<span class="sc">::</span><span class="fu">if_else</span>(</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>      quality <span class="sc">&gt;=</span> <span class="fl">0.75</span> <span class="sc">&amp;</span> <span class="co"># cells of high RF quality</span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>        <span class="sc">!</span>is_control <span class="sc">&amp;</span> <span class="co"># non-control cells</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>        cell_call <span class="sc">==</span> <span class="st">&quot;C1&quot;</span> <span class="sc">&amp;</span> <span class="co"># live cells</span></span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a>        <span class="sc">!</span>is_s_phase <span class="sc">&amp;</span> <span class="co"># that are not estimated to be in S-phase</span></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a>        total_mapped_reads <span class="sc">&gt;=</span> <span class="dv">250000</span>, <span class="co"># and have a reasonable amount of reads</span></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a>      <span class="cn">TRUE</span>,</span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a>      <span class="cn">FALSE</span></span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a>    )</span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a>  )</span></code></pre></div>
<p>Loading the reads data would be the next step. These are 500 Kb bins
where copy number estimates are converted into integer state calls by
HMMCopy.</p>
<p>Similar to the metrics, we want to mark which bins are good to use
for analyses with a few thresholds:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>reads <span class="ot">&lt;-</span> dlptools<span class="sc">::</span><span class="fu">import_dlp_files</span>(dlp_dir, <span class="st">&quot;reads&quot;</span>)</span></code></pre></div>
<p>first, let’s mark the ones that fall into regions we want to mask
(see README.md for accessing this file).</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>reads <span class="ot">&lt;-</span> dlptools<span class="sc">::</span><span class="fu">mark_mask_regions</span>(</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>  reads,</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>  <span class="at">mask_f =</span> <span class="st">&quot;../meta_data/blacklist_2023.07.17.txt&quot;</span></span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a>)</span></code></pre></div>
<p>then, we can add a column for which count as “good” bins. Again, just
sugestions with commonly used thresholds:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>reads <span class="ot">&lt;-</span> reads <span class="sc">|&gt;</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>    <span class="at">good_bin =</span> dplyr<span class="sc">::</span><span class="fu">if_else</span>(</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>      <span class="sc">!</span>mask <span class="sc">&amp;</span> <span class="co"># does not fall into mask region</span></span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>        gc <span class="sc">&gt;</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">&amp;</span> <span class="co"># bin went through GC correciton</span></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>        map <span class="sc">&gt;</span> <span class="fl">0.99</span>, <span class="co"># read bin is in a high mappability region</span></span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a>      <span class="cn">TRUE</span>,</span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a>      <span class="cn">FALSE</span></span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a>    )</span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a>  )</span></code></pre></div>
<p>Now we can combine the reads state calls with the metrics, i.e.,
contextualizing the state calls with the cell level metrics:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>reads <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>  reads, metrics,</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>  <span class="at">by =</span> <span class="st">&quot;cell_id&quot;</span>,</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a>)</span></code></pre></div>
<p>and then create a filtered set of reads for analysis</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>reads_filt <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(reads, good_cell <span class="sc">&amp;</span> good_bin)</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a></span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a><span class="co"># many columns of data available. Here is a subset</span></span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a>reads_filt[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>, <span class="dv">2</span><span class="sc">:</span><span class="dv">10</span>]</span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a><span class="co">#&gt; # A tibble: 4 × 9</span></span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a><span class="co">#&gt;     start     end chr   reads    gc   map cor_gc  copy valid</span></span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a><span class="co">#&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;fct&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;lgl&gt;</span></span>
<span id="cb9-9"><a href="#cb9-9" tabindex="-1"></a><span class="co">#&gt; 1 2000001 2500000 1       680 0.595 0.997   1.36  5.37 TRUE </span></span>
<span id="cb9-10"><a href="#cb9-10" tabindex="-1"></a><span class="co">#&gt; 2 3000001 3500000 1       642 0.585 0.997   1.27  5.02 TRUE </span></span>
<span id="cb9-11"><a href="#cb9-11" tabindex="-1"></a><span class="co">#&gt; 3 4000001 4500000 1       615 0.483 0.996   1.25  4.93 TRUE </span></span>
<span id="cb9-12"><a href="#cb9-12" tabindex="-1"></a><span class="co">#&gt; 4 4500001 5000000 1       608 0.482 0.999   1.23  4.88 TRUE</span></span></code></pre></div>
<p>A common need is to have reads summarized into segments, which are
collapsed blocks of read bins that have been assigned the same
state.</p>
<p>The DLP pipeline does produce a segments file, which you can import
with:</p>
<p><code>dlptools::import_dlp_files(dlp_dir, &quot;segs&quot;)</code></p>
<p>but that file was made with internal filters and settings that you
probably don’t know or want.</p>
<p>So we can create one from our filtered reads:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="co">#  This can take a bit to run if you have a lot of runs combined.</span></span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>segs_filt <span class="ot">&lt;-</span> dlptools<span class="sc">::</span><span class="fu">reads_to_segs_with_filters</span>(reads_filt)</span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a><span class="co"># this is now runs of adjacent read bins with identical states collapesed</span></span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a><span class="co"># into a single bin. Of course, bins are no longer of equal size.</span></span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a>segs_filt[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>, ]</span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a><span class="co">#&gt; # A tibble: 4 × 6</span></span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a><span class="co">#&gt; # Groups:   cell_id, chr [1]</span></span>
<span id="cb10-9"><a href="#cb10-9" tabindex="-1"></a><span class="co">#&gt;   cell_id                  chr      start      end state median_copy</span></span>
<span id="cb10-10"><a href="#cb10-10" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;                    &lt;fct&gt;    &lt;dbl&gt;    &lt;dbl&gt; &lt;dbl&gt;       &lt;dbl&gt;</span></span>
<span id="cb10-11"><a href="#cb10-11" tabindex="-1"></a><span class="co">#&gt; 1 AT21614-A138971B-R45-C37 1      2000001  8000000     5        4.96</span></span>
<span id="cb10-12"><a href="#cb10-12" tabindex="-1"></a><span class="co">#&gt; 2 AT21614-A138971B-R45-C37 1      8000001 10000000     7        6.71</span></span>
<span id="cb10-13"><a href="#cb10-13" tabindex="-1"></a><span class="co">#&gt; 3 AT21614-A138971B-R45-C37 1     10000001 18500000     5        5.13</span></span>
<span id="cb10-14"><a href="#cb10-14" tabindex="-1"></a><span class="co">#&gt; 4 AT21614-A138971B-R45-C37 1     18500001 23000000     7        6.57</span></span></code></pre></div>
</div>
<div id="plotting-trees-and-heatmaps" class="section level2">
<h2>Plotting Trees and Heatmaps</h2>
<p>A common plot from a DLP analysis is to build a tree with some method
(<a href="https://github.com/molonc/sitka_wrapper/tree/main">sitka</a>,
or hdbscan) and then plot it next to a heatmap of read state calls. The
function <code>dlptools::create_tree_copynumber_heatmap()</code> is
designed to do just that. It is a wrapper around several independent
steps and the full code with individual functions is available in the
repo at <code>R/plot_tree_heatmap.R</code>.</p>
<p>The point of the plot is to bring together a tree, with clone labels,
and state calls. If you have other annotations, those can be added
too.</p>
<p>Here we’ll walk through plot creation with HDBScan. This requires
installing <a href="https://github.com/shahcompbio/signals">signals</a>
in addition to this package.</p>
<p>This example is assuming you followed the steps above to create a set
of filtered reads.</p>
<p>First, need to run umap clustering and in the process create a tree
with hdbscan:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="fu">library</span>(signals)</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a>clusters <span class="ot">&lt;-</span> signals<span class="sc">::</span><span class="fu">umap_clustering</span>(</span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a>  reads_filt,</span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a>  <span class="at">minPts =</span> <span class="dv">50</span>, <span class="at">field =</span> <span class="st">&quot;state&quot;</span></span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a>)</span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a></span>
<span id="cb11-8"><a href="#cb11-8" tabindex="-1"></a><span class="co"># if you see an error like:</span></span>
<span id="cb11-9"><a href="#cb11-9" tabindex="-1"></a><span class="co">#     An error occurred in umap calculation: function &#39;as_cholmod_sparse&#39; not</span></span>
<span id="cb11-10"><a href="#cb11-10" tabindex="-1"></a><span class="co">#     provided by package &#39;Matrix&#39;</span></span>
<span id="cb11-11"><a href="#cb11-11" tabindex="-1"></a><span class="co"># you might need to install some older package versions</span></span>
<span id="cb11-12"><a href="#cb11-12" tabindex="-1"></a><span class="co"># remotes::install_version(&quot;Matrix&quot;, version = &quot;1.6-1&quot;)</span></span>
<span id="cb11-13"><a href="#cb11-13" tabindex="-1"></a><span class="co"># install.packages(&quot;irlba&quot;, type = &quot;source&quot;)</span></span>
<span id="cb11-14"><a href="#cb11-14" tabindex="-1"></a><span class="co"># https://github.com/bwlewis/irlba/issues/70</span></span></code></pre></div>
<p>Then prep the reads data (continuing with the data from example
above) for plotting:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a>reads_f_wide <span class="ot">&lt;-</span> dlptools<span class="sc">::</span><span class="fu">convert_long_reads_to_wide</span>(reads_filt)</span></code></pre></div>
<p>Now we can make the plot:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a>dlptools<span class="sc">::</span><span class="fu">create_tree_copynumber_heatmap</span>(</span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a>  <span class="at">phylo_tree =</span> clusters<span class="sc">$</span>tree,</span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a>  <span class="at">states_df =</span> reads_f_wide,</span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a>  <span class="at">clones_df =</span> clusters<span class="sc">$</span>clustering,</span>
<span id="cb13-5"><a href="#cb13-5" tabindex="-1"></a>  <span class="co"># can save direct to file by adding:</span></span>
<span id="cb13-6"><a href="#cb13-6" tabindex="-1"></a>  <span class="co"># remove to print to screen (but plots are typically large)</span></span>
<span id="cb13-7"><a href="#cb13-7" tabindex="-1"></a>  <span class="at">file_name =</span> <span class="st">&quot;../example_heatmaps/example_hdbscan.png&quot;</span></span>
<span id="cb13-8"><a href="#cb13-8" tabindex="-1"></a>)</span>
<span id="cb13-9"><a href="#cb13-9" tabindex="-1"></a><span class="co">#&gt; PNG </span></span>
<span id="cb13-10"><a href="#cb13-10" tabindex="-1"></a><span class="co">#&gt;   2</span></span></code></pre></div>
<p>here we write directly to a file. You can print to screen by not
including a <code>file_name</code> argument, but these plots are big and
are often best just dumped to a file to look at.</p>
<p><br> <br> <br></p>
<p>Lets say we had some annotation we wanted to add too (as left side
columns on the heatmap). This is done by matching sample IDs in cell_id
names, and could be read from a t(c)sv file with
<code>dlptools::import_annotations_df()</code>.</p>
<p>But for this example will just prep by hand:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a>anno_df <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a>  <span class="co"># bit of a boring example, because we only have one sample and one library</span></span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a>  <span class="at">sample_id =</span> <span class="fu">c</span>(<span class="st">&quot;AT21614&quot;</span>),</span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a>  <span class="at">passage =</span> <span class="fu">c</span>(<span class="st">&quot;p10&quot;</span>),</span>
<span id="cb14-5"><a href="#cb14-5" tabindex="-1"></a>  <span class="at">somthing_else =</span> <span class="fu">c</span>(<span class="st">&quot;featureA&quot;</span>)</span>
<span id="cb14-6"><a href="#cb14-6" tabindex="-1"></a>)</span></code></pre></div>
<p>And then we can pass this tibble to the plotting code to be
added.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a>dlptools<span class="sc">::</span><span class="fu">create_tree_copynumber_heatmap</span>(</span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a>  <span class="at">phylo_tree =</span> clusters<span class="sc">$</span>tree,</span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a>  <span class="at">states_df =</span> reads_f_wide,</span>
<span id="cb15-4"><a href="#cb15-4" tabindex="-1"></a>  <span class="at">clones_df =</span> clusters<span class="sc">$</span>clustering,</span>
<span id="cb15-5"><a href="#cb15-5" tabindex="-1"></a>  <span class="at">annos_df =</span> anno_df</span>
<span id="cb15-6"><a href="#cb15-6" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAAAM1BMVEUAAAAFe0oxgr2YAEOeyuGf/KazAADJlMfMzMzUudrdHHffZbDjSjP8jVn9yr/9zIr////hh6vQAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAMt0lEQVR4nO2ci2Ksqg6GPUuPtVM4+P5Pe7yABgwGb5jR/Ht3Op0C4reSEC62aEWrKu7uAHcJIEICiJAAIiSACAkgQpwAFYi6j/+71KLqf5Zq/7dUY0xT61KpRjfdu9K0rWmbttXdf2X/vlVt97NpQKfy3T8ljM/JgMwA6KN0ozpA5QRItbUFVL8bkB4B6RHQl1kQyud0C2qABdWlEkAYoFJ1cLp3qoYWVA+AygFQBbqVjwChPICmGFSFLlbPMUi/2IJGQKYxlfIAGXGxAVADY5CaAJkO0UcAuVHsDwU0DfMVTxfD+VwzzHcuFg7z/F0sC6A+Bildd3mQqUILqgVQ6GKDBWmQB2k3zL8WUDNYUDkAapaZtOY8zGdzsbKLQcNktVKlXmTS5dsBNYOLNVEXYztZjfC5IAaV6CjGfi6WB9DsYiOgL3KxXIB6F6u16WJQU5nJgjqjscP88EPzYkBdaDalti6m5hjU2BjE1cVifC7Lg7Sbamgw1WA8zGcCpDpAH9VPVmssSLtEUb0W0GBBdeky6XoJSL3bgkww1ViOYi93sd6ClKltDLIuVo2zeTBZ/bwaUK1L7fbFND5Zfa8FqS5KK/3Xje+9i30+ygKqoYuZ1hjQs3wQ1pQrBunFcoeNQUqG+SkGTXlQdLLKD1CUzyWAynqci5Uft3Go4GTVMMyk8wHqUkT9o/q5mDGlGojM+2K1WFAXg0rz072aYKoxzMXU6wH1iaI2fSbdL9qbyYKqIUzPC2bstp6zAaqqqs+DmqoDZOq6mXY1mvFbj6lqm/cC8s4HYaOYYbncEedzQQzqhvnxfJCpQQwyduuZZwzKB2iIQdaCRhfT/VzMbhyynWpkA6RGF6vtglntLOiX+a5GRgsat306TG7jUIM8qAP0w9GCVvhcONVovmfJNSMgNawHlX0M0pVW6PGXNwOyw3xtxsmqWrrYywH1c7Gym4v9zQeopiBdD1kix1Fsjc+liSI8H2QYu9hBQMMdbAJU2q1nBaYanGNQRkD0AaoPw32xY4CK4RZSLahqKm1UfwSvqaqmn7VX7fDS9N+UPcRZ/YPt366MgMYFs9pakCmDTPpBLuZ+N/S/2JRJf/wgDUcxnsP8Kh/SgmyhTUGaPMTJKwYdAlTY101Buh/FwsMLjM9J5wY0n3LVwMUM353VYy5mlQhoWO5Qg4v1qKbnxR4cg6w2xSDvEKcGLiaAjHUxe4izbgAgrha0zuf6RfsgSPMbxbICCh6HmiwoWHJ9tQWZ2pR2PWixsyqA+kxamz97DHge5n+CnVVGgAg+l5y0n85JxzJpThaUF9AQg9SwN9/vznsbh/AvLzDaes5vQbXbWa2gBTVc52K5ATUzoGDrmWeQpvhcNMyHo9gP22H+FkAKXe7g+VDvHYBMjT6zyvJ0B8nnfECmc7HaBNs+JdcYlBvQOFkdg3QTLLmyPN2RH1AzAAIWpBwgBR6o43LSnuZzRR5UjqNYZ0WK+6J9dkAfPajW/+oOUP07PMzyb6Q0WlCH5h8bQAl8zl5y7Qd3re2KorWgiu3Wc35Ablcj8kDdONXg8yhCfkDu8EIzTFY1GOYbaEGKR5BO4XPNVAOOYoxXFG8A5CzIPasBt33YAUricw0g+4cmZxcLNg5ZLHek8blmRVGFU43flt0fWErkc8lU4zOuKKLHgBUTC0rlk+mM4q//WPj9FpTM56IgPZyTdoAq6GI8AKXzOf9pn+F5MRuD9PjHcCYLYjKKbcBzxTlphbrYdgvadBvX6VxA/V/B0x2gesiklbeiWDtAFR2k76YCdC4ge7pDg0X7xo1in/mhXsKC7mbi6WRAfXqoa92MgJCHWTS9YHY3El9nW5Aezgf1S/cwSJvOeP4GC/oQFnQ3j4VOdzHTBWljZguy60EKjmLxree7cSx1OqA+SNsYBJ8X8/fmIxZ0NwxMVwD6g4CwJw4jgO5mgeoCQGoJqB5HsVUXu5tERPksqF63oLtBxHRVDDJeDFK0i90NIqYrXcx4z6x+vhbQvbq/B8wlgAgJIEICiNDXAOrjNR20+xJhGeyzDdfdXTOv3AOq6/0tkDLYZ1sv/AUqwNemQkkViTa/QQKIUDF72WqhdUAgAV28CX7VplyQk4oxShOF1mMQQqR1L/arHSN6MTXzNYDShE1P4Gf93ftEisKNj+P/I83HAqIE7GaGAs3J+eOUGXwPoGPpzNRGGGomMmB+XIBrfQ2glHQGXQWAGSaEA1ysWLrYvMxx0f2crbTRuvDKzh85z4F2Y00SWpAFNBb1muSv5DyoDQrCigsHWwxq4JOgTe46BdDeC3+FkmIQ+A7eH0r4vgYQmuIsysyvcxBKyTCpJh8jR/G8xeyHATpfXwNomeCslA0rpVZca4u50vp5EMb+C9+vVMs5MqLvv7AreyPNtUvPSR1KKJzEbbuNLwfkD1kooEUC9UxAkY4GgNoloOVnVwCCyUXhvs3v/O9e98E0eu375AdIiPVnTK7VaV1i/qmdfgvaBIBgjhTv9B5Ac8/mmy7mTrqvCQeo18KezzPm6TtsFX6AN+JdGVT2aoNk0b0bvrDO0ll2CiB3EXg74Y2j1/EcAJaBZdHGwkbayAWhSUQ6OMUgrLMnAsJuBfyTtREXAxaNWXZgm3NjXitzEzsAedeGVgc/uxYQvBUcECgHA0TkrpAOBFEmCmjapQjMN7wN2AHi5qkCUwcSAKHegVl120buM9JhJJitWZBvlWHn0t2rbdMAhY6A/bOBm0Dvy7UQBmkvoK40UsAy3heKDrYZ9obo9B5AwL4nH4GUpl8Ui2rAnh2R1q8DEKKNWKdxhWNmtwQELg8aIju9C9BO0QacXroArytlCr+UD2hLbw5VSm48tXXwz75ShirhB95lm18MKOnut1/dhTX3blcrJ/eKs54LqAAiCq656nMBpcpF6MhNCaACfJ3ZJiXiOYm4gocn9sl/2mK1KAYo1TvX2qQUAURXvQxQbEayyIOmD3cb1o2Aphx7vSEEEFJhTtgDekV7zPX2A0o4zrQGyKX6ZCe2AEJU2J5+CyAwgdwNaMr/7A/up2kQgwVuA5RyQQwQmAv4LzHhFrSY1rcWPlgfCJYdvgTQfNFjgGA9BwjAgi9wFr9ZewGBe4zrSkBwfQQActYFCiTfJ6rdgFJSi6stqAVQ2hDQdIcHkiCqb1YsLQj2PwXQXkI7AXn9iyp3kAY/+wEp8UYxZQcUDPO7E8Vw0XZeYS68hm8ClKaYi20SAmiLdgefDfUFUEZdfD0kHB+8Ym5A1woZW48tdjwM0NGAvNbmIySACC0AFYW4mKeD8654kw8R3B4668YeBQiVuNi6vi1RzC4BREgAERJAlGQutqIThv1nAzpBJwA6/WjYqcqxoojWK7C3XgEe3BgAivyeBaDjM4+UigV44ghsfE9/jwgeLfeqcFCOYd5uFiy2WjxMC69jAuioEi3Iu2X/XID7MYxKbwXk9sJ9QH5MZgUoQx602MWkAZ26JHNIeWIQCQgJyjz43ABocrFiBoT51EsBLYYz7oBkskpJJqvrYrUexGn0cmIFiKMEECUJ0uuSFcWLlQLonyeshPaElfh4OtTlvBJAhPYC8r06AggUwQBlygiOZR47ARV+XRwQfB4IARS0cZUK8Lq7+rp2AipaDoByLNoLIEL7ABUtAShPDGILKCghMWgByM9f7wPEeBTjYUEHJYAIXZoocgjS03NSe6snlMEA+YoBmoUByiJnp5lHsUDMAfkZx/b6j1YOQJUnrITyhJX487Szs3tUHBwNHg+onY8S7KueUAYHBGsigIJ1TgwQtw0QVHsB+feGAfLbQAA9Jw/aZ0F+GwJIAO2JQXMJAYSOYnSQFkBWsWH+vYACBxIXE0BbXYwM0s9OFH3FYtCsWAxiLwFEiL+N36wUQI0nrETtCSvx4+lQl/NKABESQIT2AvKHaAyQXwIHVCAlmWknoCDJQwAFJTBA7hG91H7comsBzVqxoEcCCqoKIAQQrIkCSo1BDwXkVcQA+ZtRrwPk19sXpJ8MKKgmgATQmjBA/nqyJIrYMA+EAfKFA/oCCSBCfG2biVIAGU9YidITVuLX06Eu55UAIrQXkD/yIICCcQ4HVATfGWonoCB3iVnQOiAHkPMofy0g0HiaBTEEdSQGCSArFBAVg4K23weopS1oIyCGf7ahvRSQ1/TbLCglSAugdUB+y28DRCeKuwAxlEw1CAkgQoyNm4cEECEBREgAERJAhAQQIQFESAAREkCEBBAhAURIABESQIQEECEBREgAERJAhAQQIQFESAAREkCEBBAhAURIABESQIQEECEBREgAERJAhAQQIQFESAAREkCEBBAhAURIABESQIQEEKH/A3EIVPDpL2yoAAAAAElFTkSuQmCC" /><!-- --></p>
<p><strong>This plot will look bad in the vignette, see the example
ouputs in the example_heatmaps/ folder.</strong></p>
<p><br> <br> <br></p>
<p>If you’ve run sitka (perhaps using the <a href="https://github.com/molonc/sitka_wrapper">sitka_wrapper</a>), you
can read in the newick tree file created and do much the same plot.
Presumably you would have also used the leiden_cut method for clone
labelling, and will need that file too.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="co"># here, pulled from the example files in the tree_cutting/ section of the</span></span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a><span class="co"># sitka_wrapper repo linked above</span></span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a>sitka_files <span class="ot">&lt;-</span> <span class="st">&quot;/projects/molonc/aparicio_lab/bfurman/code/sitka_wrapper/tree_cutting/example_inputs/&quot;</span></span>
<span id="cb16-4"><a href="#cb16-4" tabindex="-1"></a>clones_f <span class="ot">&lt;-</span> fs<span class="sc">::</span><span class="fu">path_join</span>(<span class="fu">c</span>(sitka_files, <span class="st">&quot;leiden_clones_reso010_nb10.csv&quot;</span>))</span>
<span id="cb16-5"><a href="#cb16-5" tabindex="-1"></a>sitka_reads_f <span class="ot">&lt;-</span> fs<span class="sc">::</span><span class="fu">path_join</span>(<span class="fu">c</span>(sitka_files, <span class="st">&quot;filtered_cell_state_calls.csv.gz&quot;</span>))</span>
<span id="cb16-6"><a href="#cb16-6" tabindex="-1"></a>sitka_tree_f <span class="ot">&lt;-</span> fs<span class="sc">::</span><span class="fu">path_join</span>(<span class="fu">c</span>(sitka_files, <span class="st">&quot;tree.newick&quot;</span>))</span></code></pre></div>
<p>Sitka trees need a little cleaning, so run through the tree
formatting function to drop ‘locus_’ values at tips, and clean off
‘cell_’ names from tree leafs.</p>
<p>There are also other importing functions that will help set the data
up correctly:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a></span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a><span class="co"># import and format the tree file</span></span>
<span id="cb17-3"><a href="#cb17-3" tabindex="-1"></a>sitka_tree <span class="ot">&lt;-</span> dlptools<span class="sc">::</span><span class="fu">import_tree</span>(sitka_tree_f) <span class="sc">|&gt;</span></span>
<span id="cb17-4"><a href="#cb17-4" tabindex="-1"></a>  dlptools<span class="sc">::</span><span class="fu">format_sitka_tree</span>()</span>
<span id="cb17-5"><a href="#cb17-5" tabindex="-1"></a></span>
<span id="cb17-6"><a href="#cb17-6" tabindex="-1"></a><span class="co"># expects a file with columns of: cell_id, clone_id</span></span>
<span id="cb17-7"><a href="#cb17-7" tabindex="-1"></a>clones <span class="ot">&lt;-</span> dlptools<span class="sc">::</span><span class="fu">import_clones</span>(clones_f)</span>
<span id="cb17-8"><a href="#cb17-8" tabindex="-1"></a></span>
<span id="cb17-9"><a href="#cb17-9" tabindex="-1"></a><span class="co"># standard sitka input format</span></span>
<span id="cb17-10"><a href="#cb17-10" tabindex="-1"></a>sitka_reads <span class="ot">&lt;-</span> dlptools<span class="sc">::</span><span class="fu">import_wideformat_states_file</span>(sitka_reads_f)</span>
<span id="cb17-11"><a href="#cb17-11" tabindex="-1"></a></span>
<span id="cb17-12"><a href="#cb17-12" tabindex="-1"></a><span class="co"># set up or read external annotations file with:</span></span>
<span id="cb17-13"><a href="#cb17-13" tabindex="-1"></a><span class="co"># dlptools::import_annotations_df(anno_file)</span></span>
<span id="cb17-14"><a href="#cb17-14" tabindex="-1"></a>anno_df <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb17-15"><a href="#cb17-15" tabindex="-1"></a>  <span class="co"># bit of a boring example, because we only have one sample and one library</span></span>
<span id="cb17-16"><a href="#cb17-16" tabindex="-1"></a>  <span class="at">sample_id =</span> <span class="fu">c</span>(<span class="st">&quot;AT21552&quot;</span>, <span class="st">&quot;AT21350&quot;</span>),</span>
<span id="cb17-17"><a href="#cb17-17" tabindex="-1"></a>  <span class="at">passage =</span> <span class="fu">c</span>(<span class="st">&quot;p3&quot;</span>, <span class="st">&quot;p23&quot;</span>),</span>
<span id="cb17-18"><a href="#cb17-18" tabindex="-1"></a>  <span class="at">somthing_else =</span> <span class="fu">c</span>(<span class="st">&quot;featureA&quot;</span>, <span class="st">&quot;featureB&quot;</span>)</span>
<span id="cb17-19"><a href="#cb17-19" tabindex="-1"></a>)</span></code></pre></div>
<p>and now we can create the plot:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a>dlptools<span class="sc">::</span><span class="fu">create_tree_copynumber_heatmap</span>(</span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a>  <span class="at">phylo_tree =</span> sitka_tree,</span>
<span id="cb18-3"><a href="#cb18-3" tabindex="-1"></a>  <span class="at">states_df =</span> sitka_reads,</span>
<span id="cb18-4"><a href="#cb18-4" tabindex="-1"></a>  <span class="at">clones_df =</span> clones,</span>
<span id="cb18-5"><a href="#cb18-5" tabindex="-1"></a>  <span class="at">annos_df =</span> anno_df,</span>
<span id="cb18-6"><a href="#cb18-6" tabindex="-1"></a>  <span class="at">file_name =</span> <span class="st">&quot;../example_heatmaps/example_sitka.png&quot;</span></span>
<span id="cb18-7"><a href="#cb18-7" tabindex="-1"></a>)</span>
<span id="cb18-8"><a href="#cb18-8" tabindex="-1"></a><span class="co">#&gt; PNG </span></span>
<span id="cb18-9"><a href="#cb18-9" tabindex="-1"></a><span class="co">#&gt;   2</span></span></code></pre></div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
