<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Package index â€¢ dlptools</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Nunito_Sans-0.4.9/font.css" rel="stylesheet"><link href="../deps/Noto_Sans_Mono-0.4.9/font.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Package index"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">dlptools</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.2.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/heatmaps.html">Flexible Heatmaps</a></li>
    <li><a class="dropdown-item" href="../articles/useful_functions.html">Functions in this package</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Package index</h1>
    </div>

    <div class="section level2">
      <h2 id="all-functions">All functions<a class="anchor" aria-label="anchor" href="#all-functions"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="ASCN_COLORS.html">ASCN_COLORS</a></code>

        </dt>
        <dd>colors for signals results of allele balances</dd>
      </dl><dl><dt>

          <code><a href="ASCN_PHASE_COLORS.html">ASCN_PHASE_COLORS</a></code>

        </dt>
        <dd>ASCN phase colors</dd>
      </dl><dl><dt>

          <code><a href="CNV_COLOURS.html">CNV_COLOURS</a></code>

        </dt>
        <dd>standard colors used in dlp plots</dd>
      </dl><dl><dt>

          <code><a href="DEFAULT_CONTINUOUS_COLOR_RANGE.html">DEFAULT_CONTINUOUS_COLOR_RANGE</a></code>

        </dt>
        <dd>min, median, max for a continuous color range</dd>
      </dl><dl><dt>

          <code><a href="STATE_COLORS.html">STATE_COLORS</a></code>

        </dt>
        <dd>alias of standard colors used in dlp plots</dd>
      </dl><dl><dt>

          <code><a href="basic_tile_plot.html">basic_tile_plot()</a></code>

        </dt>
        <dd>create a tile plot of read state calls.</dd>
      </dl><dl><dt>

          <code><a href="bme_vec.html">bme_vec()</a></code>

        </dt>
        <dd>picks beginning, middle, and end of a vector to handle when vectors that are too long are passed</dd>
      </dl><dl><dt>

          <code><a href="build_left_annot.html">build_left_annot()</a></code>

        </dt>
        <dd>builds the left-side annotations of the cells</dd>
      </dl><dl><dt>

          <code><a href="cell_id_order_as_plotted.html">cell_id_order_as_plotted()</a></code>

        </dt>
        <dd>grab cell ids in the order that they are plotted</dd>
      </dl><dl><dt>

          <code><a href="check_args.html">check_args()</a></code>

        </dt>
        <dd>confirms arguments are compatible for the plotting wrapper</dd>
      </dl><dl><dt>

          <code><a href="chr_name_check.html">chr_name_check()</a></code>

        </dt>
        <dd>internal for checking the names of chromosome columns in a dataframe.</dd>
      </dl><dl><dt>

          <code><a href="convert_long_reads_to_wide.html">convert_long_reads_to_wide()</a></code>

        </dt>
        <dd>convert long format reads to wide format</dd>
      </dl><dl><dt>

          <code><a href="create_chromosome_column_fct.html">create_chromosome_column_fct()</a></code>

        </dt>
        <dd>create a sorted factor vector of chromosomes</dd>
      </dl><dl><dt>

          <code><a href="factor_column_mixedsort.html">factor_column_mixedsort()</a></code>

        </dt>
        <dd>naturally sort a column from a dataframe.</dd>
      </dl><dl><dt>

          <code><a href="fetch_continuous_color_ramp.html">fetch_continuous_color_ramp()</a></code>

        </dt>
        <dd>internal function for setting up heatmap continuous range colors chooses defaults, unless overwritten by user.</dd>
      </dl><dl><dt>

          <code><a href="fetch_heatmap_color_palette.html">fetch_heatmap_color_palette()</a></code>

        </dt>
        <dd>grab colors for various hm possibilities.</dd>
      </dl><dl><dt>

          <code><a href="format_sitka_tree.html">format_sitka_tree()</a></code>

        </dt>
        <dd>clean tree tip labels and drop any locus tips from sitka trees</dd>
      </dl><dl><dt>

          <code><a href="format_states_for_hm.html">format_states_for_hm()</a></code>

        </dt>
        <dd>format states for plotting in a heatmap</dd>
      </dl><dl><dt>

          <code><a href="generate_hm_image.html">generate_hm_image()</a></code>

        </dt>
        <dd>generate a ComplexHeatmap::Heatmap image, either to console or file.</dd>
      </dl><dl><dt>

          <code><a href="generate_state_hm.html">generate_state_hm()</a></code>

        </dt>
        <dd>creates a complex heatmap of a given matrix of states.</dd>
      </dl><dl><dt>

          <code><a href="get_clone_id_label_positions.html">get_clone_id_label_positions()</a></code>

        </dt>
        <dd>get a row number of where to place each clone label.</dd>
      </dl><dl><dt>

          <code><a href="get_clone_members.html">get_clone_members()</a></code>

        </dt>
        <dd>create list of list of cell_ids that belong to each clone id.</dd>
      </dl><dl><dt>

          <code><a href="get_column_metrics.html">get_column_metrics()</a></code>

        </dt>
        <dd>get plotted values bounds</dd>
      </dl><dl><dt>

          <code><a href="import_annotations_df.html">import_annotations_df()</a></code>

        </dt>
        <dd>read an annotations file (t(c)sv)</dd>
      </dl><dl><dt>

          <code><a href="import_clones.html">import_clones()</a></code>

        </dt>
        <dd>read a clones t(c)sv file</dd>
      </dl><dl><dt>

          <code><a href="import_dlp_files.html">import_dlp_files()</a></code>

        </dt>
        <dd>generic function to import dlp files</dd>
      </dl><dl><dt>

          <code><a href="import_tree.html">import_tree()</a></code>

        </dt>
        <dd>read a tree from a file in newick format</dd>
      </dl><dl><dt>

          <code><a href="load_mask_file.html">load_mask_file()</a></code>

        </dt>
        <dd>internal to control mask file loading.</dd>
      </dl><dl><dt>

          <code><a href="make_clone_palette.html">make_clone_palette()</a></code>

        </dt>
        <dd>generate a color palette for clone labels</dd>
      </dl><dl><dt>

          <code><a href="make_corrupt_tree_heatmap.html">make_corrupt_tree_heatmap()</a></code>

        </dt>
        <dd>create a complex heatmap object with a tree.</dd>
      </dl><dl><dt>

          <code><a href="make_tree_plot_obj.html">make_tree_plot_obj()</a></code>

        </dt>
        <dd>create a ggplot object of a phylogenetic tree</dd>
      </dl><dl><dt>

          <code><a href="mark_mask_regions.html">mark_mask_regions()</a></code>

        </dt>
        <dd>add a mask column to a dataframe</dd>
      </dl><dl><dt>

          <code><a href="plot_state_hm.html">plot_state_hm()</a></code>

        </dt>
        <dd>main hm building function</dd>
      </dl><dl><dt>

          <code><a href="pull_chr_from_col_name.html">pull_chr_from_col_name()</a></code>

        </dt>
        <dd>extract chromosome name from a bin name (chr_start_end)</dd>
      </dl><dl><dt>

          <code><a href="pull_info_from_cell_id.html">pull_info_from_cell_id()</a></code>

        </dt>
        <dd>generic extractor of info contained in cell ids</dd>
      </dl><dl><dt>

          <code><a href="pull_sample_id_from_cell_id.html">pull_sample_id_from_cell_id()</a></code>

        </dt>
        <dd>extract sample ID from the typically formatted cell_ids</dd>
      </dl><dl><dt>

          <code><a href="reads_to_segs.html">reads_to_segs()</a></code>

        </dt>
        <dd>convert reads table to segments table</dd>
      </dl><dl><dt>

          <code><a href="rle_states.html">rle_states()</a></code>

        </dt>
        <dd>convert states to run length encoding</dd>
      </dl><dl><dt>

          <code><a href="sort_df_by_cell_order.html">sort_df_by_cell_order()</a></code>

        </dt>
        <dd>sort a table given a vector of cell_ids</dd>
      </dl></div>
  </main></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Ben Furman.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

