<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Flexible Heatmaps • dlptools</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Nunito_Sans-0.4.9/font.css" rel="stylesheet">
<link href="../deps/Noto_Sans_Mono-0.4.9/font.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Flexible Heatmaps">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">dlptools</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.2.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/heatmaps.html">Flexible Heatmaps</a></li>
    <li><a class="dropdown-item" href="../articles/useful_functions.html">Functions in this package</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Flexible Heatmaps</h1>
            
      

      <div class="d-none name"><code>heatmaps.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="plotting-trees-and-heatmaps">Plotting Trees and Heatmaps<a class="anchor" aria-label="anchor" href="#plotting-trees-and-heatmaps"></a>
</h2>
<p>DLP data is commonly vizualized as a heatmap of copy number states
across the genome.</p>
<p>Next to this heatmap, we often want to add:</p>
<ol style="list-style-type: decimal">
<li>phylogenetic trees of the cellular relationships</li>
<li>annotations of sample, experiments</li>
<li>clonal identities both as labels and colors on the tree</li>
</ol>
<p>The package <a href="https://github.com/shahcompbio/signals/tree/master" class="external-link">Signals</a>
has great functions to do this (and lots of other useful functions for
DLP and single cell analyses).</p>
<p>Alternatively, in this package there is
<code><a href="../reference/plot_state_hm.html">dlptools::plot_state_hm()</a></code>, which does similar things to the
Signals package, but with some more conveniences.</p>
<p><code><a href="../reference/plot_state_hm.html">dlptools::plot_state_hm()</a></code> is meant to be a one stop shop
for heatmap plotting of many variable types and options.</p>
<p><code>?dlptools::plot_state_hm()</code> will show the options, and
below we’ll walk throuh some examples.</p>
<p>The main constraint is to have a column called <code>cell_id</code>
that contains the cell labels.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dlptools</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_state_hm.html">plot_state_hm</a></span><span class="op">(</span></span>
<span>  <span class="co"># long format bin level data</span></span>
<span>  <span class="va">states_df</span>,</span>
<span>  <span class="co"># column of data to plot</span></span>
<span>  <span class="va">state_col</span>,</span>
<span>  <span class="co"># optional, tree with tip labels matching the cell_id column</span></span>
<span>  phylogeny <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="co"># for direct saving to a file</span></span>
<span>  file_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="co"># optional dataframe of a cell_id column + any cell level information columns</span></span>
<span>  anno_df <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="co"># for custom colors of annotations, see ?plot_state for examples</span></span>
<span>  anno_colors_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  <span class="co"># can call out the annotation columns from the original dataframe</span></span>
<span>  <span class="co"># instead of a separate frame.</span></span>
<span>  anno_columns <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="co"># optional, can also specify columns in the dataframe</span></span>
<span>  clones_df <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="co"># column name to pull clone IDs from instead of separate DF.</span></span>
<span>  clone_column <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="co"># bool, color the tree with clone colors</span></span>
<span>  color_tree_clones <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="co"># bool, only letter label the largest group of any given clone</span></span>
<span>  only_largest_clone_group <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="co"># font size of labels on heatmap image</span></span>
<span>  labels_fontsize <span class="op">=</span> <span class="fl">8</span>,</span>
<span>  <span class="co"># set to TRUE if plotting continuous values on the heatmap.</span></span>
<span>  continuous_hm_colours <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="co"># specificiation of color range for continuous values</span></span>
<span>  custom_continuous_colors <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="co"># specification of values for ends of continous range.</span></span>
<span>  custom_continuous_range <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>We’ll work from some example data (a trimmed output of signals, but
just DLP reads output works fine too):</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ex_state_dat</span> <span class="op">&lt;-</span> <span class="fu">vroom</span><span class="fu">::</span><span class="fu"><a href="https://vroom.r-lib.org/reference/vroom.html" class="external-link">vroom</a></span><span class="op">(</span><span class="st">"data/ex_state_dat.tsv.gz"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ex_state_dat</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 11</span></span></span>
<span><span class="co">#&gt;   cell_id sample_id passage   chr  start    end state   BAF state_AS state_phase</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> AT2399… AT23998   p1          1 2.00<span style="color: #949494;">e</span>6 2.5 <span style="color: #949494;">e</span>6     4 0.179 3|1      A-Gained   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> AT2399… AT23998   p1          1 3.00<span style="color: #949494;">e</span>6 3.5 <span style="color: #949494;">e</span>6     4 0.294 3|1      A-Gained   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> AT2399… AT23998   p1          1 4.00<span style="color: #949494;">e</span>6 4.50<span style="color: #949494;">e</span>6     4 0.211 3|1      A-Gained   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> AT2399… AT23998   p1          1 4.50<span style="color: #949494;">e</span>6 5   <span style="color: #949494;">e</span>6     5 0.299 5        A-Gained   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> AT2399… AT23998   p1          1 5.00<span style="color: #949494;">e</span>6 5.50<span style="color: #949494;">e</span>6     5 0.182 5        A-Gained   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> AT2399… AT23998   p1          1 5.50<span style="color: #949494;">e</span>6 6   <span style="color: #949494;">e</span>6     5 0.129 5        A-Gained   </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 1 more variable: copy &lt;dbl&gt;</span></span></span></code></pre></div>
<div class="section level3">
<h3 id="basic-heatmap">basic heatmap<a class="anchor" aria-label="anchor" href="#basic-heatmap"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dlptools</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_state_hm.html">plot_state_hm</a></span><span class="op">(</span></span>
<span>  states_df <span class="op">=</span> <span class="va">ex_state_dat</span>,</span>
<span>  state_col <span class="op">=</span> <span class="st">"state"</span>,</span>
<span>  <span class="co"># optional, but recommended dump direct to a file with:</span></span>
<span>  file_name <span class="op">=</span> <span class="st">"imgs/basic_hm.png"</span></span>
<span>  <span class="co"># recommended for full, large, heatmaps</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="imgs%2Fbasic_hm.png" width="75%" height="75%"></p>
<p>Or with a different column:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dlptools</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_state_hm.html">plot_state_hm</a></span><span class="op">(</span></span>
<span>  states_df <span class="op">=</span> <span class="va">ex_state_dat</span>,</span>
<span>  state_col <span class="op">=</span> <span class="st">"state_phase"</span>,</span>
<span>  <span class="co"># optional, but recommended dump direct to a file with:</span></span>
<span>  file_name <span class="op">=</span> <span class="st">"imgs/basic_phase_hm.png"</span></span>
<span>  <span class="co"># recommended for full, large, heatmaps</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="imgs%2Fbasic_phase_hm.png" width="75%" height="75%"></p>
<p>Column options include:</p>
<ol style="list-style-type: decimal">
<li>state</li>
<li>state_phase (A-gained, B-hom, etc)</li>
<li>A</li>
<li>B</li>
<li>BAF</li>
<li>state_AS</li>
<li>state_AS_phased (might be too many states to usefully
visualize)</li>
<li>or pretty much any custom column you want to plot that is reasonably
similar.</li>
</ol>
<hr>
</div>
<div class="section level3">
<h3 id="adding-a-tree">Adding a Tree<a class="anchor" aria-label="anchor" href="#adding-a-tree"></a>
</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ex_tree</span> <span class="op">&lt;-</span> <span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/read.tree.html" class="external-link">read.tree</a></span><span class="op">(</span><span class="st">"data/pkg_tree.newick"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">dlptools</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_state_hm.html">plot_state_hm</a></span><span class="op">(</span></span>
<span>  states_df <span class="op">=</span> <span class="va">ex_state_dat</span>,</span>
<span>  state_col <span class="op">=</span> <span class="st">"state"</span>,</span>
<span>  phylogeny <span class="op">=</span> <span class="va">ex_tree</span>,</span>
<span>  file_name <span class="op">=</span> <span class="st">"imgs/with_tree_hm.png"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="imgs%2Fwith_tree_hm.png" width="75%" height="75%"></p>
</div>
<div class="section level3">
<h3 id="adding-annotation-data">Adding Annotation Data<a class="anchor" aria-label="anchor" href="#adding-annotation-data"></a>
</h3>
<p>This can come from a separate data frame with annotations per cell
ID, or you can point to columns in your state dataframe:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dlptools</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_state_hm.html">plot_state_hm</a></span><span class="op">(</span></span>
<span>  states_df <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span><span class="va">ex_state_dat</span>, passage <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">passage</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  state_col <span class="op">=</span> <span class="st">"state"</span>,</span>
<span>  file_name <span class="op">=</span> <span class="st">"imgs/annotations_hm.png"</span>,</span>
<span>  anno_columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sample_id"</span>, <span class="st">"passage"</span><span class="op">)</span>,</span>
<span>  <span class="co"># you could control the colors of your annotations with:</span></span>
<span>  <span class="co"># anno_colors_list = list(passage=c(`1`='#2872bc', `19`='#d23e3e')),</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="imgs%2Fannotations_hm.png" width="75%" height="75%"></p>
<p>Or with some pre-made annotation dataframe created by cell id:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">anno_df</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="va">ex_state_dat</span>, <span class="va">cell_id</span>, <span class="va">sample_id</span>, <span class="va">passage</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="fu">dlptools</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_state_hm.html">plot_state_hm</a></span><span class="op">(</span></span>
<span>  states_df <span class="op">=</span> <span class="va">ex_state_dat</span>,</span>
<span>  state_col <span class="op">=</span> <span class="st">"state"</span>,</span>
<span>  anno_df <span class="op">=</span> <span class="va">anno_df</span>,</span>
<span>  file_name <span class="op">=</span> <span class="st">"imgs/annotations_2_hm.png"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="imgs%2Fannotations_2_hm.png" width="75%" height="75%"></p>
<p>And with either of these, we can add a tree with the
<code>phylogeny</code> arguments.</p>
</div>
<div class="section level3">
<h3 id="clone-information">Clone Information<a class="anchor" aria-label="anchor" href="#clone-information"></a>
</h3>
<p>Clones work similar to annotations, where you can either supply a
data frame with <code>cell_id</code> and <code>clone_id</code> columns,
or just pull the information from the states dataframe:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># fake some clone data</span></span>
<span><span class="va">ex_state_dat</span> <span class="op">&lt;-</span> <span class="va">ex_state_dat</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span></span>
<span>    clone <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/if_else.html" class="external-link">if_else</a></span><span class="op">(</span><span class="va">passage</span> <span class="op">==</span> <span class="st">"p1"</span>, <span class="st">"A"</span>, <span class="st">"B"</span><span class="op">)</span>,</span>
<span>  <span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="fu">dlptools</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_state_hm.html">plot_state_hm</a></span><span class="op">(</span></span>
<span>  states_df <span class="op">=</span> <span class="va">ex_state_dat</span>,</span>
<span>  state_col <span class="op">=</span> <span class="st">"state_phase"</span>,</span>
<span>  file_name <span class="op">=</span> <span class="st">"imgs/with_clones.png"</span>,</span>
<span>  clone_column <span class="op">=</span> <span class="st">"clone"</span>,</span>
<span>  <span class="co"># optional, don't have to have annotations, or could pass the dataframe like</span></span>
<span>  <span class="co"># above</span></span>
<span>  anno_columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sample_id"</span>, <span class="st">"passage"</span><span class="op">)</span>,</span>
<span>  <span class="co"># optional, don't have to have tree</span></span>
<span>  phylogeny <span class="op">=</span> <span class="va">ex_tree</span>,</span>
<span>  <span class="co"># optional, turns on tree coloring by clone</span></span>
<span>  color_tree_clones <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="co"># optional, only largest cell group of a clone gets a letter label.</span></span>
<span>  only_largest_clone_group <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="imgs%2Fwith_clones.png" width="75%" height="75%"></p>
</div>
<div class="section level3">
<h3 id="continous-variables">Continous Variables<a class="anchor" aria-label="anchor" href="#continous-variables"></a>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dlptools</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_state_hm.html">plot_state_hm</a></span><span class="op">(</span></span>
<span>  states_df <span class="op">=</span> <span class="va">ex_state_dat</span>,</span>
<span>  state_col <span class="op">=</span> <span class="st">"copy"</span>,</span>
<span>  continuous_hm_colours <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="co"># optional, can specify the colors</span></span>
<span>  <span class="co"># custom_continuous_colors = c("#000000", "#ffffff", "#5F9EA0")</span></span>
<span>  <span class="co"># optional, can specify values to fill out low, mid, high end of range</span></span>
<span>  <span class="co"># basically has the effect of squishing or stretching the color scale</span></span>
<span>  <span class="co"># custom_continuous_range = c(0, 2, 10)</span></span>
<span>  file_name <span class="op">=</span> <span class="st">"imgs/continuous.png"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># and all of this can be specified with annotations, trees, etc.</span></span>
<span><span class="co"># as above.</span></span></code></pre></div>
<p><img src="imgs%2Fcontinuous.png" width="75%" height="75%"></p>
</div>
<div class="section level3">
<h3 id="errors">Errors<a class="anchor" aria-label="anchor" href="#errors"></a>
</h3>
<p>If you see an error like this:</p>
<pre><code>Error in `fortify()`:
! `data` must be a &lt;data.frame&gt;, or an object coercible by `fortify()`, or a valid &lt;data.frame&gt;-like object coercible by `as.data.frame()`.
Caused by error in `.prevalidate_data_frame_like_object()`:
! `dim(data)` must return an &lt;integer&gt; of length 2.</code></pre>
<p>That’s a complaint from ggtree that I need to fix and haven’t
yet.</p>
<p>Loading ggtree should fix it:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://www.amazon.com/Integration-Manipulation-Visualization-Phylogenetic-Computational-ebook/dp/B0B5NLZR1Z/" class="external-link">ggtree</a></span><span class="op">)</span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Ben Furman.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
